### ISSUE TITLE
String interpolation is not working with sh command

### ISSUE THREAD

**Issue by elferherrera** [nushell/nushell#3668]
**Describe the bug**

When using string interpolation with an external command, the variable is not found by nushell

**To Reproduce**
1. let foo = abc
2. dd if=/dev/zero bs=1M count=1 $"of=($foo)"

**Expected behavior**
Run command `dd` using `foo` in the argument

**Screenshots**
![image](https://user-images.githubusercontent.com/6942205/122984806-5552a700-d395-11eb-96a6-4686ed116338.png)

**Configuration (please complete the following information):**

Run `version | pivot` and paste the output to show OS, features, etc.

```
> version | pivot
────┬────────────────────┬───────────────────────────────────────────────────────────────────────────────────────────────────────────────
 #  │      Column0       │                                                    Column1                                                    
────┼────────────────────┼───────────────────────────────────────────────────────────────────────────────────────────────────────────────
  0 │ version            │ 0.33.0                                                                                                        
  1 │ branch             │ main                                                                                                          
  2 │ short_commit       │ 55cab9eb                                                                                                      
  3 │ commit_hash        │ 55cab9eb4ff4ee3ee73efc6f8973901b2a91c921                                                                      
  4 │ commit_date        │ 2021-06-22 05:22:33                                                                                           
  5 │ build_os           │ linux-x86_64                                                                                                  
  6 │ rust_version       │ rustc 1.53.0 (53cb7b09b 2021-06-17)                                                                           
  7 │ rust_channel       │ stable-x86_64-unknown-linux-gnu (default)                                                                     
  8 │ cargo_version      │ cargo 1.53.0 (4369396ce 2021-04-27)                                                                           
  9 │ pkg_version        │ 0.33.0                                                                                                        
 10 │ build_time         │ 2021-06-22 13:11:44                                                                                           
 11 │ build_rust_channel │ release                                                                                                       
 12 │ features           │ clipboard-cli, ctrlc, default, rustyline, term, which, zip                                                    
 13 │ installed_plugins  │ chart bar, chart line, fetch, from sqlite, inc, match, post, ps, s3, selector, sys, textview, to sqlite, tree 
────┴────────────────────┴───────────────────────────────────────────────────────────────────────────────────────────────────────────────
```

**Comment by schrieveslaach** [comment]
Confirmed. I have the same error in #3596. I'll try to fix it.

**Comment by schrieveslaach** [comment]
I put a log expression into `crates/nu-engine/src/evaluate/evaluator.rs` and compared the output between 0.32 and 0.33:

```rust
pub fn evaluate_baseline_expr(
    expr: &SpannedExpression,
    ctx: &EvaluationContext,
) -> Result<Value, ShellError> {
    log::error!(":::::: {:?} :::::", expr);
```

The difference is always that external commands receive always `FullColumnPath` as an `SpannedExpression` expression.

### 0.32.0

```
❯ ^ls $"(pwd)"
 ERROR nu_engine::evaluate::evaluator > :::::: SpannedExpression { expr: Invocation(Block { params: Signature { name: "<invocation>", usage: "", positional: [], rest_positional: None, named: {"help": (Switch(Some('h')), "Display this help message")}, yields: None, input: None, is_filter: false }, block: [Group { pipelines: [Pipeline { list: [Internal(InternalCommand { name: "build-string", name_span: Span { start: 4, end: 12 }, args: Call { head: SpannedExpression { expr: Synthetic(String("build-string")), span: Span { start: 4, end: 12 } }, positional: Some([SpannedExpression { expr: Invocation(Block { params: Signature { name: "<basic>", usage: "", positional: [], rest_positional: None, named: {"help": (Switch(Some('h')), "Display this help message")}, yields: None, input: None, is_filter: false }, block: [Group { pipelines: [Pipeline { list: [Internal(InternalCommand { name: "pwd", name_span: Span { start: 7, end: 10 }, args: Call { head: SpannedExpression { expr: Command, span: Span { start: 7, end: 10 } }, positional: None, named: Some(NamedArguments { named: {"help": AbsentSwitch} }), span: Span { start: 7, end: 10 }, external_redirection: None } })], span: Span { start: 7, end: 10 } }], span: Span { start: 0, end: 0 } }], definitions: {}, span: Span { start: 0, end: 0 } }), span: Span { start: 7, end: 11 } }]), named: None, span: Span { start: 4, end: 12 }, external_redirection: Stdout } })], span: Span { start: 4, end: 12 } }], span: Span { start: 4, end: 12 } }], definitions: {}, span: Span { start: 4, end: 12 } }), span: Span { start: 4, end: 12 } } :::::
 ERROR nu_engine::evaluate::evaluator > :::::: SpannedExpression { expr: Invocation(Block { params: Signature { name: "<basic>", usage: "", positional: [], rest_positional: None, named: {"help": (Switch(Some('h')), "Display this help message")}, yields: None, input: None, is_filter: false }, block: [Group { pipelines: [Pipeline { list: [Internal(InternalCommand { name: "pwd", name_span: Span { start: 7, end: 10 }, args: Call { head: SpannedExpression { expr: Command, span: Span { start: 7, end: 10 } }, positional: None, named: Some(NamedArguments { named: {"help": AbsentSwitch} }), span: Span { start: 7, end: 10 }, external_redirection: None } })], span: Span { start: 7, end: 10 } }], span: Span { start: 0, end: 0 } }], definitions: {}, span: Span { start: 0, end: 0 } }), span: Span { start: 7, end: 11 } } :::::
CODE_OF_CONDUCT.md  Cargo.toml	   README.build.txt  debian  extra_features_cargo_install.sh  rustfmt.toml  target
CONTRIBUTING.md     LICENSE	   README.md	     docker  images			      samples	    tests
Cargo.lock	    Makefile.toml  crates	     docs    pkg_mgrs			      src	    wix
 ERROR nu_engine::evaluate::evaluator > :::::: SpannedExpression { expr: Literal(String("prompt")), span: Span { start: 9, end: 15 } } :::::
```

### 0.33.0

```
❯ ^ls $"(pwd)"
 ERROR nu_engine::evaluate::evaluator > :::::: SpannedExpression { expr: Literal(String("ls")), span: Span { start: 1, end: 3 } } :::::
 ERROR nu_engine::evaluate::evaluator > :::::: SpannedExpression { expr: FullColumnPath(FullColumnPath { head: SpannedExpression { expr: Variable("$\"(pwd)\"", Span { start: 4, end: 12 }), span: Span { start: 4, end: 12 } }, tail: [] }), span: Span { start: 4, end: 12 } } :::::
 ERROR nu_engine::evaluate::evaluator > :::::: SpannedExpression { expr: Variable("$\"(pwd)\"", Span { start: 4, end: 12 }), span: Span { start: 4, end: 12 } } :::::
error: Variable not in scope
  ┌─ shell:1:5
  │
1 │ ^ls $"(pwd)"
  │     ^^^^^^^^ unknown variable: $"(pwd)"

 ERROR nu_engine::evaluate::evaluator > :::::: SpannedExpression { expr: Literal(String("starship")), span: Span { start: 0, end: 8 } } :::::
 ERROR nu_engine::evaluate::evaluator > :::::: SpannedExpression { expr: Literal(String("prompt")), span: Span { start: 9, end: 15 } } :::::
```

**Comment by voanhduy1512** [comment]
The problem happened after #3642.
